File: src/242.valid-anagram.ts
/*
 * @lc app=leetcode id=242 lang=typescript
 *
 * [242] Valid Anagram
 */

// @lc code=start
function isAnagram(s: string, t: string): boolean {
  if (s.length != t.length) return false
  const freq = new Array<number>(26).fill(0)
  for (let i = 0; i < s.length; i++) {
    // 97 = "a"
    freq[s.charCodeAt(i) - 97]++
    freq[t.charCodeAt(i) - 97]--
  }
  return freq.every((val) => val == 0)
}
// @lc code=end

File: src/242.valid-anagram.go
/*
 * @lc app=leetcode id=242 lang=golang
 *
 * [242] Valid Anagram
 */

// @lc code=start
// package main

func isAnagram(s string, t string) bool {
	if len(s) != len(t) {
		return false
	}
	hashmap := make(map[rune]int)
	for i := 0; i < len(s); i++ {
		hashmap[rune(s[i])] = hashmap[rune(s[i])] + 1
		hashmap[rune(t[i])] = hashmap[rune(t[i])] - 1
	}
	for _, value := range hashmap {
		if value != 0 {
			return false
		}
	}
	return true
}

// func main() {
// 	isAnagram("abcd", "dacb")
// }

// @lc code=end
